File: Creational/Builder/Java/AbstractBuilder.java

public abstract class AbstractBuilder {
	
	protected AbstractProduct abstractProduct;

	public AbstractProduct getProduct() {
		return this.abstractProduct;
	}
	
	public abstract void reset();
	public abstract void buildStepA();
	public abstract void buildStepB();
	public abstract void buildStepC();
}

File: Creational/Builder/Python/AbstractBuilder.py
from abc import ABC, abstractmethod

class AbstractBuilder(ABC):

	def get_product(self):
		return self.product
		
	@abstractmethod
	def reset(self):
		pass
		
	@abstractmethod
	def build_step_a(self):
		pass
		
	@abstractmethod
	def build_step_b(self):
		pass
		
	@abstractmethod
	def build_step_c(self):
		pass